python_modules/libraries/dagster-k8s/dagster_k8s/utils.py

```diff
@@ -10,6 +12,13 @@
 )
 
 
+def sanitize_k8s_label(label_name: str):
+    # Truncate too long label values to fit into 63-characters limit and avoid invalid characters.
+    # https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
+    label_name = label_name[:63]
+    return re.sub("[^a-z0-9-]", "", label_name).strip("-")
```

Comment:
`_` and `.` pass the regex for k8s, probably worth keeping those

what about replacing with `-` or something instead of blanking it out? just makes cases like the dynamic output a bit hard to read